doc/whats_new/v0.20.rst

```diff
@@ -130,6 +141,10 @@ Metrics
 - :func:`metrics.roc_auc_score` now supports binary ``y_true`` other than
   ``{0, 1}`` or ``{-1, 1}``. :issue:`9828` by :user:`Hanmin Qin <qinhanmin2014>`.
 
+- :func:`metrics.cluster.silhouette_score` and
+  :func:`metrics.cluster.silhouette_samples` are more memory efficient,
+  causing them to run faster. :issue:`10280` by `Joel Nothman`_.
```

Comment:
The latter part? silhouette_samples rather than silhouette_score? All the work is done in silhouette_samples... There are some basic benchmarks at https://github.com/scikit-learn/scikit-learn/pull/7177, but I should run them again and more thoroughly.

Reply:
Yes, thanks for linking the benchmarks. The speedup does look impressive but I'm wondering if the speedup there is also due to some re-write of the code and not just chunking.

Reply:
I meant about `silhouette_samples` running faster.

Reply:
Again, I should run the benchmarks again, but it *should* be due to the chunking as the bottleneck in the previous implementation was clearly in the pairwise distance calculation.